// ts-password-generator/dist/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generator = void 0;
const lowToHigh = (low, high) => {
    const array = [];
    for (let i = low; i <= high; i++) {
        array.push(i);
    }
    return array;
};
const lowercaseCharactersCode = lowToHigh(97, 122);
const uppercaseCharactersCode = lowToHigh(65, 90);
const numberCharactersCode = lowToHigh(48, 57);
const symbolsCharactersCode = lowToHigh(33, 47)
    .concat(lowToHigh(58, 64))
    .concat(lowToHigh(123, 126))
    .concat(lowToHigh(91, 96));
const generator = ({ charsQty = 10, isUppercase = false, haveString = true, haveNumbers = false, haveSymbols = false, } = {
    charsQty: 10,
    isUppercase: false,
    haveString: true,
    haveNumbers: false,
    haveSymbols: false,
}) => {
    let charactersCode = [];
    if (haveString)
        charactersCode = charactersCode.concat(lowercaseCharactersCode);
    if (isUppercase)
        charactersCode = charactersCode.concat(uppercaseCharactersCode);
    if (haveNumbers)
        charactersCode = charactersCode.concat(numberCharactersCode);
    if (haveSymbols)
        charactersCode = charactersCode.concat(symbolsCharactersCode);
    const password = [];
    if (charactersCode.length === 0) {
        return '';
    }
    for (let i = 0; i < charsQty; i++) {
        const char = Math.floor(Math.random() * charactersCode.length);
        password.push(String.fromCharCode(charactersCode[char]));
    }
    return password.join('');
};
exports.generator = generator;
